docs/design-proposals/2022-04 TraceQL Concepts.md

```diff
@@ -0,0 +1,203 @@
+`{ duration > 2s }`
+
+Find traces that contain a span named "HTTP POST":  
+`{ name = "HTTP POST" }`
+
+### Attribute fields
+
+We can refer to dynamic attributes (also known as tags) on the span or the span's resource.
+
+**Examples**
+
+Find traces with the GET http method:  
+`{ http.method = "GET" }`
+
+Find traces that passed through the prod namespace:  
+`{ namespace = "prod" }`
+
+Find traces that cross service boundaries:  
+`{ parent.service.name != service.name }`
+
+### Field expressions
+
+Fields can also be combined in various expected ways.
+
+**Examples**
+
+Find traces where the difference of the duration of a set of parent/child spans exceeds 500ms:  
+`{ parent.duration - duration > 500ms }`
+
+Find traces with "success" http status codes:  
+`{ http.status >= 200 && http.status < 300 }`
+
+Note that the second expression requires both conditions to be true on the same span. The entire expression inside of `{}` must be evaluated as true on a single span for it to be included in the resultset.
+
+## Combining Spansets
+
+Logical operators combine sets of spans. For example, if we wanted to find a trace that went through two specific regions:  
+`{ region = "eu-west-0" } && { region = "eu-west-1" } `
+
+Note the difference between the above and the following:  
+`{ region = "eu-west-0"  && region = "eu-west-1" } `
+
+The second expression will return no traces because it's impossible for both conditions to be simultaneously true on the same span.
+
+### Structural operators
+
+Structural operators evaluate traces based on the span relationships within the span tree. 
+
+`{ } >> { }`  
+  This is the descendant operator. The spans returned from this operator will match the right hand side conditions while also being descendants of spans that match the left hand side conditions.
+
+`{ } > { }`  
+  This is the child operator. The spans returned from this operator will match the right hand side conditions while also being children of spans that match the left hand side conditions.
+
+`{ } ~ { }`  
+  This is the sibling operator. The spans returned from this operator will match the right hand side conditions while also being siblings of spans that match the left hand side conditions.
+
+## Aggregators
+
+All of the above expressions involve asking questions about individual spans. However, sometimes we want to ask questions about a set of spans. For that we have aggregate functions.
+
+**Examples**
+
+Find traces where the total number of spans is greater than 10:  
+`count() > 10`
+
+Find traces where the average duration is greater than 1s:  
+`avg(duration) > 1s`
+
+## Expression Pipelining
+
+Pipelining allows us to "pipe" a set of spans from one expression to the next. This is particularly useful if you want to perform an aggregate over a subset of a trace.
+
+**Examples**
+
+Find traces that have more than 3 spans with an attribute `http.status` with a value of `200`:  
+`{ http.status = 200 } | count() > 3`
```

Comment:
That answers my question about the mysql example :-) I'll leave it there, just to confirm that this was indeed the logical outcome.